spec/inputs/sqs_spec.rb

```diff
@@ -201,7 +201,63 @@ def poll(polling_options = {})
         end
       end
 
-      context "other errors" do
+      context "SQS error (retries)" do
+
+        it "retry to fetch messages" do
+          sleep_time = LogStash::Inputs::SQS::BACKOFF_SLEEP_TIME
+          expect(subject).to receive(:sleep).with(sleep_time)
+          expect(subject).to receive(:sleep).with(sleep_time * 2)
+          expect(subject).to receive(:sleep).with(sleep_time * 4)
+
+          error_count = 0
+          expect(mock_sqs).to receive(:poll).with(anything()).at_most(4) do
+            error_count += 1
+            if error_count <= 3
+              raise Aws::SQS::Errors::QueueDoesNotExist.new("testing", "testing exception (#{error_count})")
+            end
+
+            queue << payload
+          end
+
+          subject.run(queue)
+
+          expect(queue.size).to eq(1)
+          expect(queue.pop).to eq(payload)
+        end
+
+      end
+
+      context "networking error" do
+
+        before(:all) { require 'seahorse/client/networking_error' }
+
+        it "retry to fetch messages" do
+          sleep_time = LogStash::Inputs::SQS::BACKOFF_SLEEP_TIME
+          expect(subject).to receive(:sleep).with(sleep_time).twice
+
+          error_count = 0
+          expect(mock_sqs).to receive(:poll).with(anything()).at_most(5) do
+            error_count += 1
+            if error_count == 1
+              raise Seahorse::Client::NetworkingError.new(Net::OpenTimeout.new, 'timeout')
+            end
+            if error_count == 3
```

Comment:
Is this combining two test scenarios? One, that the backoff resets after a successful run, and two, testing the `NetworkingError` class of events? Maybe split these into separate specs?

Reply:
yeah it is as I also wanted to test the `sleep_time` being reset after a successful invocation